components/net/filemanager_thread.rs

```diff
@@ -0,0 +1,145 @@
+                        None => {
+                            let _ = sender.send(Err(FileManagerThreadError::InvalidSelection));
+                            return;
+                        }
+                    }
+                }
+
+                let _ = sender.send(Ok(replies));
+            },
+            SelectedPort::Single(sender) => {
+                let selected_path = "";
+
+                match self.create_entry(selected_path) {
+                    Some(triple) => {
+                        let _ = sender.send(Ok(triple));
+                    },
+                    None => {
+                        let _ = sender.send(Err(FileManagerThreadError::InvalidSelection));
+                    }
+                }
+            }
+        }
+    }
+
+    fn create_entry(&mut self, file_path: &str) -> Option<(Uuid, PathBuf, u64)> {
+        match File::open(file_path) {
+            Ok(handler) => {
+                let id = Uuid::new_v4();
+                self.idmap.borrow_mut().insert(id, file_path.to_string());
+
+                // Unix Epoch: https://doc.servo.org/std/time/constant.UNIX_EPOCH.html
+                let epoch = handler.metadata().and_then(|metadata| metadata.modified()).map_err(|_| ())
+                                              .and_then(|systime| systime.elapsed().map_err(|_| ()))
+                                              .and_then(|elapsed| {
+                                                    let secs = elapsed.as_secs();
+                                                    let nsecs = elapsed.subsec_nanos();
+                                                    let msecs = secs * 1000 + nsecs as u64 / 1000000;
+                                                    Ok(msecs)
+                                                });
+
+                let filename = Path::new(file_path).file_name();
+
+                match (epoch, filename) {
+                    (Ok(epoch), Some(filename)) => Some((id, Path::new(filename).to_path_buf(), epoch)),
+                    _ => None
+                }
+            },
+            Err(_) => None
+        }
+    }
+
+    fn read_file(&mut self, sender: IpcSender<FileManagerResult<Vec<u8>>>, id: Uuid) {
+
+        match self.idmap.borrow().get(&id).and_then(|filepath| {
+            let mut buffer = vec![];
+            match File::open(&filepath) {
+                Ok(mut handler) => {
+                    match handler.read_to_end(&mut buffer) {
+                        Ok(_) => Some(buffer),
+                        Err(_) => None,
+                    }
+                },
+                Err(_) => None,
+            }
+        }) {
+            Some(buffer) => {
+                let _ = sender.send(Ok(buffer));
+            },
+            None => {
+                let _ = sender.send(Err(FileManagerThreadError::ReadFileError));
+            }
+        };
+    }
+
+    fn delete_fileid(&mut self, id: Uuid) {
```

Comment:
Nah, just consistency. This isn't really important, so feel free to keep it as is if you like this name.